granulate_utils/linux/ns.py

```diff
@@ -59,9 +59,13 @@ def resolve_proc_root_links(proc_root: str, ns_path: str) -> str:
     parts = Path(ns_path).parts
 
     path = proc_root
+    seen = set()
     for part in parts[1:]:  # skip the / (or multiple /// as .parts gives them)
         next_path = os.path.join(path, part)
-        if os.path.islink(next_path):
+        while os.path.islink(next_path):
```

Comment:
`path = next_path` also needs to be in the loop, I think.

Reply:
Seems so

Reply:
I think you're right that what I did won't work for relative links, but I'm not sure `path = next_path` in the loop will fix it. I think maybe instead of using `path` for the relative case we should use `os.path.dirname(next_path)`, wdyt?

Reply:
There's no need to get rid of `..`. The purpose of this function is to prepare a `/proc/pid/root` path that "the kernel will eat". Our only problem is with absolute symlinks - relative ones, incl `..` I suppose, work just fine.

Did you test it with a final path containing `..` and the resulting path turned out inaccessible?

Reply:
I didn't have a specific problem with having `..` in the path, just thought that we can remove it like the corresponding `os.path.realpath` that does the same thing sans mounts. I'll remove it

Reply:
Current code lgtm, resolving this